src/System/Net/CertificateHelper.cs

```diff
@@ -0,0 +1,233 @@
+
+        /// <summary>
+        /// Issue a chain-signed SSL certificate with private key.
+        /// </summary>
+        /// <param name="certSubject">Certificate subject (domain name).</param>
+        /// <param name="issuerCertificate">Authority's certificate used to sign this certificate.</param>
+        /// <param name="serverIp">IP Address of the remote server.</param>
+        /// <param name="certHashAlgorithm">Certificate hash algorithm.</param>
+        /// <param name="certVaildBeforeNow">Minimum Certificate validity Date.</param>
+        /// <param name="certVaildAfterNow">Maximum Certificate validity Date.</param>
+        /// <param name="wildcard">(optional) Enables wildcard SAN attributes.</param>
+        /// <returns>Signed chain of SSL Certificates.</returns>
+        public static X509Certificate MakeChainSignedCert(string certSubject, X509Certificate2 issuerCertificate, HashAlgorithmName certHashAlgorithm,
+            IPAddress serverIp, DateTimeOffset certVaildBeforeNow, DateTimeOffset certVaildAfterNow, bool wildcard = false)
+        {
+            // Look if it is already issued.
+            // Why: https://support.mozilla.org/en-US/kb/Certificate-contains-the-same-serial-number-as-another-certificate
+            if (FakeCertificates.ContainsKey(certSubject))
+            {
+                X509Certificate2 CachedCertificate = FakeCertificates[certSubject];
+                //check that it hasn't expired
+                if (CachedCertificate.NotAfter > DateTime.Now && CachedCertificate.NotBefore < DateTime.Now)
+                { return CachedCertificate; }
+                else
+                { FakeCertificates.Remove(certSubject, out _); }
+            }
+
+            using RSA issuerPrivKey = issuerCertificate.GetRSAPrivateKey() ?? throw new Exception("Issuer Certificate doesn't have a private key, Chain Signed Certificate will not be generated.");
+
+            // If not found, initialize private key generator & set up a certificate creation request.
+            using RSA rsa = RSA.Create();
+
+            // Generate an unique serial number.
+            byte[] certSerialNumber = new byte[16];
+            new Random().NextBytes(certSerialNumber);
+
+            // set up a certificate creation request.
+            CertificateRequest certRequestAny = new CertificateRequest($"CN={certSubject} [{GetRandomInt64(100, 999)}], OU=SpaceStation14 Department," +
+                $" O=\"SpaceWizards Corp\", L=New York, S=Northeastern United, C=US", rsa, certHashAlgorithm, RSASignaturePadding.Pkcs1);
+
+            // set up a optional SAN builder.
+            SubjectAlternativeNameBuilder sanBuilder = new SubjectAlternativeNameBuilder();
+
+            sanBuilder.AddDnsName(certSubject); // Some legacy clients will not recognize the cert serial-number.
+            sanBuilder.AddEmailAddress("SpaceWizards@gmail.com");
```

Comment:
This email address does not exist and it should not be hardcoded in the library.